EagleThreadBot/Program.cs

```diff
@@ -34,19 +33,67 @@ internal static async Task Main(String[] args)
 			});
 
 			await Client.ConnectAsync();
+			await UpdateLocalCache();
 
+			StartTimer();
+			
 			Slashies = Client.UseSlashCommands();
 
 			Slashies.RegisterCommands<CreateCommand>(Configuration.GuildId);
 			Slashies.RegisterCommands<TagCommand>(Configuration.GuildId);
 			Slashies.RegisterCommands<SuggestCommand>(Configuration.GuildId);
+			Slashies.RegisterCommands<TagsCommand>(Configuration.GuildId);
+			Slashies.RegisterCommands<CacheCommand>(Configuration.GuildId);
 
 			Slashies.SlashCommandErrored += Slashies_SlashCommandErrored;
 
 			await Task.Delay(-1);
 		}
 
-		private async static Task Slashies_SlashCommandErrored(SlashCommandsExtension sender, SlashCommandErrorEventArgs e)
+        private static void StartTimer()
+        {
+			Int32 timerInterval = 15;
+			if (Configuration.UpdateCache < timerInterval)
+				timerInterval = 15 * 60 * 1000;
+			else
+				timerInterval = Configuration.UpdateCache * 60 * 1000;
+
+			timer.Elapsed += Timer_Elapsed;
+			timer.Interval = timerInterval;
+			timer.AutoReset = true;
+			timer.Enabled = true;
+			timer.Start();
+		}
+
+        private static async void Timer_Elapsed(object sender, ElapsedEventArgs e)
+        {
+			await UpdateLocalCache();
+        }
+
+        private static async Task UpdateLocalCache()
+        {
+			if (File.Exists("./cache/index.json")
+				&& File.GetLastWriteTimeUtc("./cache/index.json") 
+				>= (DateTime.UtcNow + TimeSpan.FromMinutes(15)))
+				return;
+
+			if (!Directory.Exists("./cache"))
+				Directory.CreateDirectory("./cache");
+			if (!File.Exists("./cache/index.json"))
+				File.Create("./cache/index.json");
+			
+			String index = await httpClient.GetStringAsync($"{Program.Configuration.TagUrl}index.json");
+
+			String cache = File.ReadAllText("./cache/index.json");
+
+			if (cache != index)
+			{
+				File.WriteAllText("./cache/index.json", index);
+				Client.Logger.LogInformation(new EventId(10, "Cache"), "Cache updated");
+			}
```

Comment:
just force write, there's no point in checking for equality. its less performance intensive to just overwrite a file entirely than to read it and check equality of a potentially very long string.

Reply:
fair enough, done